include/stumpless/param.h

```diff
@@ -447,6 +447,35 @@ STUMPLESS_PUBLIC_FUNCTION
 const char *
 stumpless_param_to_string( const struct stumpless_param *param );
 
+/**
+ * Copies the name and the value from param to a buffer string.
+ *
+ * **Thread Safety: MT-Safe**
+ * This function is thread safe. A mutex is used to coordinate the read of the
+ * param with other accesses and modifications.
+ *
+ * **Async Signal Safety: AS-Unsafe lock heap**
+ * This function is not safe to call from signal handlers due to the use of a
+ * non-reentrant lock to coordinate access and the use of memory management
+ * functions to create the result.
+ *
+ * **Async Cancel Safety: AC-Unsafe lock heap**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled, due to the use of a lock that could be left locked as well as
+ * memory management functions.
+ *
+ * @since release v3.0.0
+ *
+ * @param param The param to get the name and the value from.
+ *
+ * @return The size written to the buffer string if no error is
+ * encountered. If an error is encountered, then minimum required size of
+ * the buffer is returned.
```

Comment:
Also specify that an error code is set in the error scenario, and that the minimum required size is not returned in event of some errors (like a NULL arg).